File: BusinessCardHolder/BusinessCard.cpp
// BusinessCard.cpp
#include "BusinessCard.h"

BusinessCard::BusinessCard()
{

}

BusinessCard::BusinessCard(string companyName, string address, string companyTelephoneNumber, string faxNumber, string url,
	string personalName, string personalTelephoneNumber, string emailAddress, string position)
	:company(companyName, address, companyTelephoneNumber, faxNumber, url),
	personal(personalName, personalTelephoneNumber, emailAddress, position)
{

}

BusinessCard::BusinessCard(const BusinessCard& source)
	:company(source.company), personal(source.personal)
{

}

BusinessCard::~BusinessCard()
{

}

bool BusinessCard::IsEqual(const BusinessCard& other)
{
	bool ret = false;

	if (this->company.IsEqual(other.company) == true &&	
		this->personal.IsEqual(other.personal) == true)
	{
		ret = true;
	}

	return ret;
}

bool BusinessCard::IsNotEqual(const BusinessCard& other)
{
	bool ret = false;

	if (this->company.IsNotEqual(other.company) ==true ||
		this->personal.IsNotEqual(other.personal) == true)
	{
		ret = true;
	}

	return ret;
}

bool BusinessCard::operator ==(const BusinessCard& other)
{
	bool ret = false;

	if (this->company == other.company && this->personal == other.personal)
	{
		ret = true;
	}

	return ret;
}

bool BusinessCard::operator !=(const BusinessCard& other)
{
	bool ret = false;

	if (this->company != other.company || this->personal != other.personal)
	{
		ret = true;
	}

	return ret;
}

BusinessCard& BusinessCard::operator =(const BusinessCard& source)
{
	this->company = source.company;
	this->personal = source.personal;

	return *this;
}

File: BusinessCardHolder/Java/BusinessCard.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package businesscardholder;

/**
 *
 * @author Taehoo
 */
public class BusinessCard {
    
    public BusinessCard(){
    }
    public BusinessCard(String companyName, String address, String companyTelephoneNumber,
            String faxNumber, String url, String personalName, String personalTelephoneNumber,
            String emailAddress, String position) {
        this.company = new Company(companyName, address, companyTelephoneNumber, faxNumber, url);
        this.personal = new Personal(personalName, personalTelephoneNumber, emailAddress, position);
    }
    public boolean IsEqual(BusinessCard other){
        boolean ret= false;
        if(this.company.IsEqual(other.company) == true &&
               this.personal.IsEqual(other.personal) == true ){
            ret = true;
        }
        return ret;
    }
    public boolean IsNotEqual(BusinessCard other){
        boolean ret= false;
        if(this.company.IsNotEqual(other.company) == true &&
               this.personal.IsNotEqual(other.personal) == true ){
            ret = true;
        }
        return ret;
    }
    
     public String GetCompanyName(){
        return this.company.GetName();
    }
    public String GetAddress(){
        return this.company.GetAddress();
    }
    public String GetCompanyTelephoneNumber(){
        return this.company.GetTelephoneNumber();
    }
    public String GetFaxNumber(){
        return this.company.GetFaxNumber();
    }
    public String GetUrl(){
        return this.company.GetUrl();
    }
    public String GetPersonalName(){
        return this.personal.GetName();
    }
    public String GetPersonalTelephoneNumber(){
        return this.personal.GetTelephoneNumber();
    }
    public String GetEmailAddress(){
        return this.personal.GetEmailAddress();
    }
    public String GetPosition(){
        return this.personal.GetPosition();
    }
    
    
    private Company company;
    private Personal personal;
}
